[Bug]: Examples do not run

### Minimal, reproducible example

```TypeScript
#!/usr/bin/env bash
git clone git@github.com:puppeteer/puppeteer.git
cd puppeteer
npm i
NODE_PATH=../ node examples/search.js
# I added a GitHub Action in my fork of the project showing the error:
# https://github.com/douglasnaphas/puppeteer/blob/mod-not-found-in-examples/.github/workflows/examples.yml
# https://github.com/douglasnaphas/puppeteer/actions/runs/6594749468/job/17918974510
```


### Error string

no error

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Background

I am trying to run the examples in the directory https://github.com/puppeteer/puppeteer/tree/main/examples/.

### Expectation

I expect the example `examples/search.js` to run successfully when I follow the steps stated in examples/README.md. Those steps are given in my reproducible example.

### Reality

I get an error:

```
~/repos/puppeteer $ NODE_PATH=../ node examples/search.js
node:internal/modules/cjs/loader:571
      throw e;
      ^

Error: Cannot find module '/Users/douglasnaphas/repos/puppeteer/node_modules/puppeteer/lib/cjs/puppeteer/puppeteer.js'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1096:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1089:15)
    at resolveExports (node:internal/modules/cjs/loader:565:14)
    at Module._findPath (node:internal/modules/cjs/loader:634:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1061:27)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/Users/douglasnaphas/repos/puppeteer/examples/search.js:24:19)
    at Module._compile (node:internal/modules/cjs/loader:1254:14) {
  code: 'MODULE_NOT_FOUND',
  path: '/Users/douglasnaphas/repos/puppeteer/node_modules/puppeteer/package.json'
}

Node.js v18.16.0
```

The above is the error I get on my MacBook.

I added a GitHub Action in my fork of this project, so you can see the error happening.

https://github.com/douglasnaphas/puppeteer/blob/mod-not-found-in-examples/.github/workflows/examples.yml
https://github.com/douglasnaphas/puppeteer/actions/runs/6594749468/job/17918974510

![Screenshot 2023-10-20 at 10 46 30 PM](https://github.com/puppeteer/puppeteer/assets/5433410/4163804f-e2c4-4117-9784-bec22099eecc)
![Screenshot 2023-10-20 at 10 47 00 PM](https://github.com/puppeteer/puppeteer/assets/5433410/ffba6579-55cf-4749-83be-88f04604cd52)


### Puppeteer configuration file (if used)

_No response_

### Puppeteer version

v21.4.0

### Node version

v18.16.0

### Package manager

npm

### Package manager version

9.5.1

### Operating system

macOS

--- a/examples/README.md
+++ b/examples/README.md
@@ -1,6 +1,6 @@
 # Running the examples
 
-Assuming you have a checkout of the Puppeteer repo and have run npm i (or yarn) to install the dependencies, the examples can be run from the root folder like so:
+Assuming you have a checkout of the Puppeteer repo and have run `npm i` (or `yarn`) to install the dependencies, and `npm run build` (or `yarn run build`) to build the project, the examples can be run from the root folder like so:
 
 ```bash
 NODE_PATH=../ node examples/search.js

